### PR TITLE
{ARM} Bump azure-mgmt-resource from v2019-07-01 to v2020-06-01

### PR DESCRIPTION
**Pull request by zhoxing-ms**
**Description**  

Bump azure-mgmt-resource from v2019-07-01 to v2020-06-01
For PR: #13429 #14241
At the same time, some compatibility logic was made for the breaking change of SDK to ensure that breaking will not be produced on the CLI side, such as: [click](https://github.com/Azure/azure-cli/pull/14344#discussion_r459841349)

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/pull/14344#discussion_r457927719).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S173

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -5973,12 +5984,16 @@ def create_vpn_connection(cmd, resource_group_name, connection_name, vnet_gatewa
     deployment_name = 'vpn_connection_deploy_' + random_string(32)
     client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES).deployments
     properties = DeploymentProperties(template=template, parameters=parameters, mode='incremental')
+    deployment = Deployment(properties=properties)
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:5987]: Is it a breaking change?

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:5987]: Yes, it is a breaking change of SDK, and we did made some compatibility logic to ensure that the CLI will not be broken

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -1054,16 +1054,19 @@ def _invoke_deployment(cli_ctx, resource_group_name, deployment_name, template,
...
                                   subscription_id=subscription_id).deployments
     if validate:
         logger.info('==== BEGIN TEMPLATE ====')
         logger.info(json.dumps(template, indent=2))
         logger.info('==== END TEMPLATE ====')
-        return smc.validate(resource_group_name, deployment_name, properties)
-    return sdk_no_wait(no_wait, smc.create_or_update, resource_group_name, deployment_name, properties)
+        validation_poller = smc.validate(resource_group_name, deployment_name, deployment)
+        return LongRunningOperation(cli_ctx)(validation_poller)
+    return sdk_no_wait(no_wait, smc.create_or_update, resource_group_name, deployment_name, deployment)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:1069]: Compatibility logic for breaking change from SDK

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 3]
```diff
@@ -426,34 +434,42 @@ def validate_arm_template_at_subscription_scope(cmd,
...
         raise CLIError(err_message)
     if validate_only:
         return validation_result
 
-    return sdk_no_wait(no_wait, mgmt_client.create_or_update_at_subscription_scope,
-                       deployment_name, deployment_properties, deployment_location)
+    if cmd.supported_api_version(min_api='2019-10-01'):
+        return sdk_no_wait(no_wait, mgmt_client.create_or_update_at_subscription_scope, deployment_name, deployment)
+    return sdk_no_wait(no_wait, mgmt_client.create_or_update_at_subscription_scope, deployment_name,
+                       deployment_properties, deployment_location)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py:472]: Compatibility logic for breaking change from SDK

**Comment by zhoxing-ms** [comment]
@samkreter Could you please help to review this PR for your part~

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features_prop.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#Wed Jul 22 11:25:50 Pacific Daylight Time 2020
+#Fri Jul 24 17:56:48 China Standard Time 2020
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features_prop.json:1]: Does this comments affect test result?

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features_prop.json:1]: This is automatically generated and should not affect the test results

**Pull request merged**